components/dropdown/dropdown-content-mixin.js

```diff
@@ -0,0 +1,556 @@
+				reflect: true,
+				attribute: 'min-width'
+			},
+			maxWidth: {
+				type: Number,
+				reflect: true,
+				attribute: 'max-width'
+			},
+			noAutoClose: {
+				type: Boolean,
+				reflect: true,
+				attribute: 'no-auto-close'
+			},
+			noAutoFit: {
+				type: Boolean,
+				reflect: true,
+				attribute: 'no-auto-fit'
+			},
+			noAutoFocus: {
+				type: Boolean,
+				reflect: true,
+				attribute: 'no-auto-focus'
+			},
+			noPadding: {
+				type: Boolean,
+				reflect: true,
+				attribute: 'no-padding'
+			},
+			noPointer: {
+				type: Boolean,
+				reflect: true,
+				attribute: 'no-pointer'
+			},
+			align: {
+				type: String,
+				reflect: true
+			},
+			boundary: {
+				type: Object,
+			},
+			opened: {
+				type: Boolean,
+				reflect: true
+			},
+			/**
+			 * Private.
+			 */
+			openedAbove: {
+				type: Boolean,
+				reflect: true,
+				attribute: 'opened-above'
+			},
+			/**
+			 * Whether to render the content immediately. By default, the content rendering
+			 * is deferred.
+			 */
+			renderContent: {
+				type: Boolean,
+				attribute: 'render-content'
+			},
+			verticalOffset: {
+				type: String,
+				attribute: 'vertical-offset'
+			}
+		};
+	}
+
+	constructor() {
+		super();
+
+		this.__opened = false;
+		this.__content = null;
+		this.__isRTL = false;
+		this.__previousFocusableAncestor = null;
+		this.__applyFocus = true;
+		this.__dismissibleId = null;
+
+		this.__onResize = this.__onResize.bind(this);
+		this.__onAutoCloseFocus = this.__onAutoCloseFocus.bind(this);
+		this.__onAutoCloseClick = this.__onAutoCloseClick.bind(this);
+		this.__toggleScrollStyles = this.__toggleScrollStyles.bind(this);
+	}
+
+	get opened() {
+		return this.__opened;
+	}
+
+	set opened(val) {
+		const oldVal = this.__opened;
+		this.__opened = val;
+		this.requestUpdate('opened', oldVal);
+		this.__openedChanged(val);
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+
+		this.__content = this.__getContentContainer();
+		this.__content.addEventListener('scroll', this.__toggleScrollStyles);
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+
+		window.addEventListener('resize', this.__onResize);
+		document.body.addEventListener('focus', this.__onAutoCloseFocus, true);
+		document.body.addEventListener('click', this.__onAutoCloseClick, true);
+
+		this.addEventListener('d2l-dropdown-close', this.__onClose);
+		this.addEventListener('d2l-dropdown-position', this.__toggleScrollStyles);
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		window.removeEventListener('resize', this.__onResize);
+		document.body.removeEventListener('focus', this.__onAutoCloseFocus, true);
+		document.body.removeEventListener('click', this.__onAutoCloseClick, true);
+		this.removeEventListener('d2l-dropdown-close', this.__onClose);
+		this.removeEventListener('d2l-dropdown-position', this.__toggleScrollStyles);
+		this.__content.removeEventListener('scroll', this.__toggleScrollStyles);

```

Comment:
Note: there would be no need to remove this handler here since it is currently wired up in `firstUpdated` and would be cleaned up when the node is garbage collected since it is attached to a node in this element's tree.  If the component were disconnected and then reconnected (ex. moved in the DOM) this would currently result in broken scroll behavior because the handler would not be re-wired in the connected callback.  However, per previous comment, the wire-up can be moved into the handler... either way, we should be able to remove this line.